spec/lib/breadcrumbs_on_rails/json_ld/builder_spec.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe BreadcrumbsOnRails::JsonLd::Builder, type: :helper do
+  subject { helper.render_breadcrumbs(builder: described_class) }
+
+  let(:expectation) do
+    JSON.pretty_generate(
+      "@context": "http://schema.org",
+      "@type": "BreadcrumbList",
+      "itemListElement": expected_item_list_element,
+    )
+  end
+
+  context "when #add_breadcrumb is not called in advance" do
+    let(:expected_item_list_element) { [] }
+    it { is_expected.to eq expectation }
+  end
+
+  context "when #add_breadcrumb is called in advance" do
+    before do
+      helper.add_breadcrumb("root", :root_path)
+      helper.add_breadcrumb("foo", proc { |c| c.foo_path })
+      helper.add_breadcrumb("bar", "/foo/bar")
+    end
+
+    let(:expected_item_list_element) do
+      [
+        {
+          "@type": "ListItem",
+          position: 1,
+          item: {
+            "@id": "http://test.host/",
+            name: "root",
+          },
+        },
+        {
+          "@type": "ListItem",
+          position: 2,
+          item: {
+            "@id": "http://test.host/foo",
+            name: "foo",
+          },
+        },
+        {
+          "@type": "ListItem",
+          position: 3,
+          item: {
+            "@id": "http://test.host/foo/bar",
```

Comment:
[Q]
In Rails 4.0, this test fails because `URI::HTTP` instances are serialized to an object like 

```json
{"scheme":"http","user":null,"password":null,"host":"test.host","port":80,"path":"/foo/bar","query":null,"opaque":null,"fragment":null,"parser":{"regexp":{"SCHEME":"(?-mix:\\A[A-Za-z][A-Za-z0-9+\\-.]*\\z)","USERINFO":"(?-mix:\\A(?:%\\h\\h|[!$\u0026-.0-;=A-Z_a-z~])*\\z)","HOST":"(?-mix:\\A(?:(?\u003CIP-literal\u003E\\[(?:(?\u003CIPv6address\u003E(?:\\h{1,4}:){6}(?\u003Cls32\u003E\\h{1,4}:\\h{1,4}|(?\u003CIPv4address\u003E(?\u003Cdec-octet\u003E[1-9]\\d|1\\d{2}|2[0-4]\\d|25[0-5]|\\d)\\.\\g\u003Cdec-octet\u003E\\.\\g\u003Cdec-octet\u003E\\.\\g\u003Cdec-octet\u003E))|::(?:\\h{1,4}:){5}\\g\u003Cls32\u003E|\\h{,4}::(?:\\h{1,4}:){4}\\g\u003Cls32\u003E|(?:(?:\\h{1,4}:)?\\h{1,4})?::(?:\\h{1,4}:){3}\\g\u003Cls32\u003E|(?:(?:\\h{1,4}:){,2}\\h{1,4})?::(?:\\h{1,4}:){2}\\g\u003Cls32\u003E|(?:(?:\\h{1,4}:){,3}\\h{1,4})?::\\h{1,4}:\\g\u003Cls32\u003E|(?:(?:\\h{1,4}:){,4}\\h{1,4})?::\\g\u003Cls32\u003E|(?:(?:\\h{1,4}:){,5}\\h{1,4})?::\\h{1,4}|(?:(?:\\h{1,4}:){,6}\\h{1,4})?::)|(?\u003CIPvFuture\u003Ev\\h+\\.[!$\u0026-.0-;=A-Z_a-z~]+))\\])|\\g\u003CIPv4address\u003E|(?\u003Creg-name\u003E(?:%\\h\\h|[!$\u0026-.0-9;=A-Z_a-z~])*))\\z)","ABS_PATH":"(?-mix:\\A\\/(?:%\\h\\h|[!$\u0026-.0-;=@-Z_a-z~])*(?:\\/(?:%\\h\\h|[!$\u0026-.0-;=@-Z_a-z~])*)*\\z)","REL_PATH":"(?-mix:\\A(?:%\\h\\h|[!$\u0026-.0-;=@-Z_a-z~])+(?:\\/(?:%\\h\\h|[!$\u0026-.0-;=@-Z_a-z~])*)*\\z)","QUERY":"(?-mix:\\A(?:%\\h\\h|[!$\u0026-.0-;=@-Z_a-z~\\/?])*\\z)","FRAGMENT":"(?-mix:\\A(?:%\\h\\h|[!$\u0026-.0-;=@-Z_a-z~\\/?])*\\z)","OPAQUE":"(?-mix:\\A(?:[^\\/].*)?\\z)","PORT":"(?-mix:\\A[\\x09\\x0a\\x0c\\x0d ]*\\d*[\\x09\\x0a\\x0c\\x0d ]*\\z)"}}}
```

I think there are 2 options to solve the problem:

* Drop support Rails 4.0!
* Apply the following patch to `Jsonld::Builder`:

```diff
@@ -20,7 +20,7 @@ module BreadcrumbsOnRails
           "@type" => "ListItem",
           "position" => index,
           "item" => {
-            "@id" => URI.join(@context.root_url, compute_path(element)),
+            "@id" => URI.join(@context.root_url, compute_path(element)).to_s,
             "name" => compute_name(element),
           },
         }
```

Both of them seem reasonable to me. What do you think?

Reply:
@yasaichi 

> Drop support of Rails 4.0!

Sounds like music, haha.

but the latter apparently doesn't hurt anyone,
so, why don't we add `to_s` at the end of the statement?